tests/test_plugins.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import annotations
```

Comment:
There were some tests that reproduce `logfire` behavior. I think it would be useful porting those to this implementation.

Reply:
I don't think we need this now, we can do that (via another simple package probably) in future.

Reply:
the pytest shim stuff isn't necessarily testing the same stuff - we really need to write packages to file.

We could use `tmp_path`, create the different files, then run python code.